mixer/adapter/stackdriver/contextgraph/workload_test.go

```diff
@@ -143,3 +148,199 @@ func TestWorkloadInstanceClusterLocation(t *testing.T) {
 		}
 	}
 }
+
+func TestTrafficAssertionReify(t *testing.T) {
+
+	cases := []struct {
+		name         string
+		assertion    trafficAssertion
+		wantEdges    []edge
+		wantEntities []entity
+	}{
+		{"http-service",
+			trafficAssertion{
+				source:             istioPolicyWorkloadInstance,
+				destination:        istioTelemetryWorkloadInstance,
+				contextProtocol:    "http",
+				destinationService: svc,
+				timestamp:          time.Now(),
+			},
+			[]edge{
+				// policy memberships
+				edge{policyWorkloadEntity.fullName, policyOwnerEntity.fullName, membershipTypeName},
+				edge{policyOwnerEntity.fullName, policyWorkloadInstanceEntity.fullName, membershipTypeName},
+
+				// policy -> k8s membership
+				edge{policyOwnerEntity.fullName,
+					"//container.googleapis.com/projects/org2:project2/locations//clusters/global-mesh/" +
+						"k8s/namespaces/istio-system/extensions/deployments/istio-policy",
+					membershipTypeName},
+				edge{policyWorkloadInstanceEntity.fullName,
+					"//container.googleapis.com/projects/org2:project2/locations//clusters/global-mesh/" +
+						"k8s/namespaces/istio-system/pods/istio-policy-65db5b46fc-r7qhq",
+					membershipTypeName},
+
+				// policy -> service comms
+				edge{policyWorkloadInstanceEntity.fullName, svcEntity.fullName, httpComm},
+				edge{policyOwnerEntity.fullName, svcEntity.fullName, httpComm},
+				edge{policyWorkloadEntity.fullName, svcEntity.fullName, httpComm},
+
+				// telemetry memberships
+				edge{telemetryWorkloadEntity.fullName, telemetryOwnerEntity.fullName, membershipTypeName},
+				edge{telemetryOwnerEntity.fullName, telemetryWorkloadInstanceEntity.fullName, membershipTypeName},
+
+				// telemetry -> k8s membership
+				edge{telemetryOwnerEntity.fullName,
+					"//container.googleapis.com/projects/org2:project2/locations//clusters/global-mesh/" +
```

Comment:
Not related to this PR, but do we right now assume context graph adapter only works for one cluster? Seems like all metadata comes from metadata server, do we need to rethink or extend edge template when multi-cluster is supported?

Reply:
When we have cluster / mesh attributes, this will probably need to be accounted for in the template, yes.